js/src/common/states/ModalState.js

```diff
@@ -0,0 +1,14 @@
+export default class ModalState {
+  constructor(modalClass, attrs = {}) {
+    this.attrs = attrs;
+    this.modalClass = modalClass;
+  }
+
+  getAttrs() {
+    return this.attrs;
+  }
+
+  getClass() {
+    return this.modalClass;
+  }
+}
```

Comment:
Hmm, this is exactly the same as the class (and the same purpose) as the `PageState` in #2156. And very, very similar to the `AlertState` in #2163.

What is this really? Maybe a "frozen component"?

Reply:
It's to allow different component classes for different modals. wdym by frozen component?